naga/src/front/wgsl/parse/mod.rs

```diff
@@ -2472,6 +2472,9 @@ impl Parser {
                 } else {
                     Err(Error::DiagnosticAttributeNotSupported {
                         on_what_plural,
+                        // In this case the user may have intended to create a global diagnostic filter directive,
+                        // so display a note to them suggesting the correct syntax.
+                        intended_diagnostic_directive: on_what_plural == "semicolons",
```

Comment:
nitpick: I think it makes more sense to use an `enum` to track these cases, and move the "rendering" of those cases to a string into either an `impl` block or the `match` statement building a `ParseError`.

I'll push up what I'm thinking here in a bit.